src/Tasks/Microsoft.NET.Build.Tasks/GetPackagesToPrune.cs

```diff
@@ -276,6 +251,66 @@ static Dictionary<string, NuGetVersion> LoadPackagesToPruneFromTargetingPack(Log
+                packages = LoadPackagesToPruneFromFrameworkPackages(targetFrameworkIdentifier, targetFrameworkVersion, frameworkReference);
+                if (packages != null)
+                {
+                    log.LogMessage("Loaded prune package data from framework packages");
+                }
+                else
+                {
+                    log.LogMessage("Failed to load prune package data from framework packages");
+                }
+            }
+            else
+            {
+                log.LogMessage("Loading prune package data from PrunePackageData folder");
+                packages = LoadPackagesToPruneFromPrunePackageData(targetFrameworkIdentifier, targetFrameworkVersion, frameworkReference, prunePackageDataRoot);
+
+                //  For the version of the runtime that matches the current SDK version, we don't include the prune package data in the PrunePackageData folder.  Rather,
+                //  we can load it from the targeting packs that are packaged with the SDK.
+                if (packages == null)
+                {
+                    log.LogMessage("Failed to load prune package data from PrunePackageData folder, loading from targeting packs instead");
+                    packages = LoadPackagesToPruneFromTargetingPack(log, targetFrameworkIdentifier, targetFrameworkVersion, frameworkReference, targetingPackRoots);
+                }
+
+                //  Fall back to framework packages data for older framework for WindowsDesktop if necessary
+                //  https://github.com/dotnet/windowsdesktop/issues/4904
+                if (packages == null && frameworkReference.Equals("Microsoft.WindowsDesktop.App", StringComparison.OrdinalIgnoreCase))
+                {
+                    log.LogMessage("Failed to load prune package data for WindowsDesktop from targeting packs, loading from framework packages instead");
+                    packages = LoadPackagesToPruneFromFrameworkPackages(targetFrameworkIdentifier, targetFrameworkVersion, frameworkReference,
+                        acceptNearestMatch: true);
+                }
+            }
+
+            //  If LoadPrunePackageDataFromNearestFramework is true and we still haven't found data, try the previous framework version
+            if (packages == null && loadPrunePackageDataFromNearestFramework)
+            {
+                // If we can go to a lower version, recursively try it
+                if (targetVersion.Major > MinSupportedFrameworkMajorVersion)
+                {
+                    string fallbackVersion = $"{targetVersion.Major - 1}.0";
+                    log.LogMessage($"LoadPrunePackageDataFromNearestFramework is enabled, trying to load from framework version {fallbackVersion}");
+                    packages = TryLoadPackagesToPruneForVersion(log, targetFrameworkIdentifier, fallbackVersion, frameworkReference, targetingPackRoots, prunePackageDataRoot, loadPrunePackageDataFromNearestFramework);
```

Comment:
Do we want this to be recursive?  That means someone can stack overflow if someone has this setting enabled and targets something like net5000.0?  This also would be pretty expensive to fallback through each version 1-by-1...  Find nearest might be better if you can determine which frameworks you have data for.

Reply:
You'll get an error way before this target runs if you target a higher version of .NET than the SDK supports:

> NETSDK1045: The current .NET SDK does not support targeting .NET 20.0.  Either target .NET 10.0 or lower, or use a version of the .NET SDK that supports .NET 20.0. Download the .NET SDK from https://aka.ms/dotnet/download

Recursion seemed simpler to me in this case.